client/changelog/changelog-dialog.tsx

```diff
@@ -62,45 +63,22 @@ const ChangelogLoadable = React.lazy(async () => {
   return { default: () => <Content dangerouslySetInnerHTML={{ __html: html }} /> }
 })
 
-export default class ChangelogDialog extends React.Component {
-  _setting = false
-
-  componentDidMount() {
-    window.addEventListener('storage', this.onStorageChange)
-  }
-
-  componentWillUnmount() {
-    window.removeEventListener('storage', this.onStorageChange)
-  }
-
-  render() {
-    return (
-      <Dialog
-        title={"What's new"}
-        onCancel={this.onDismiss}
-        showCloseButton={true}
-        dialogRef={this.props.dialogRef}>
-        <React.Suspense fallback={<LoadingDotsArea />}>
-          <ChangelogLoadable />
-        </React.Suspense>
-      </Dialog>
-    )
-  }
-
-  onStorageChange = e => {
-    if (!this._setting && e.key === KEY) {
-      if (!shouldShowChangelog()) {
-        this.onDismiss()
-      }
+export function ChangelogDialog(props: CommonDialogProps) {
+  useEffect(() => {
+    if (shouldShowChangelog()) {
```

Comment:
this check should be unnecessary I think

Reply:
Yeah, the only reason I left it is because people can open the changelog manually whenever; but if writing to the local storage is not an expensive operation, might as well always do it.

Reply:
It's not like this is happening every frame, it happens once when the dialog opens

Reply:
This also involves a local storage read to even do this check, so it's not really "free" if trying to avoid expensive operations was even necessary here